redbox-core/redbox/chains/graph.py

```diff
@@ -100,19 +100,25 @@ def set_prompt_args(state: ChainState):
 
 
 def build_llm_chain(
-    llm: BaseChatModel,
     tokeniser: Encoding,
-    llm_max_tokens: int,
     final_response_chain=False,
 ) -> Runnable:
-    _llm = llm.with_config(tags=["response_flag"]) if final_response_chain else llm
-    return RunnableParallel(
-        {
-            "response": make_chat_prompt_from_messages_runnable(tokeniser=tokeniser, llm_max_tokens=llm_max_tokens)
-            | _llm
-            | StrOutputParser(),
-        }
-    )
+    @chain
+    def _build_llm_chain(chain_state: ChainState):
+        llm = get_llm(chain_state["query"].ai_settings)
+        _llm = llm.with_config(tags=["response_flag"]) if final_response_chain else llm
+
+        return RunnableParallel(
+            {
+                "response": make_chat_prompt_from_messages_runnable(
+                    tokeniser=tokeniser, llm_max_tokens=chain_state["query"].ai_settings.llm_max_tokens
```

Comment:
if max tokens is in the state can't we just get it inside this function and avoid passing it separately?